### PR TITLE
Minor change to support for the Arduino Zero's SAMD21G18A processor

### PR DESCRIPTION
**Pull request by jake-b**
Conditional compile #ifdefs to support the Arduino Zero's SAMD21G18A processor.

### CODE REVIEW

File: Adafruit_SSD1306.cpp [hunk 1]
```diff
@@ -192,10 +192,17 @@ void Adafruit_SSD1306::begin(uint8_t vccstate, uint8_t i2caddr, bool reset) {
...
     // Force 400 KHz I2C, rawr! (Uses pins 20, 21 for SDA, SCL)
     TWI1->TWI_CWGR = 0;
     TWI1->TWI_CWGR = ((VARIANT_MCK / (2 * 400000)) - 4) * 0x101;
+#elif defined(__SAMD21G18A__)
+    // Last I checked, Wire.setClock is a dummy function on the Arduino Zero.
+    // This seems to work instead, but maybe there's a better way?
+    // Configure 400khz I2C
+    sercom3.disableWIRE();  // BAUD is an ENABLE-protected register.  Disable it frst
+    SERCOM3->I2CM.BAUD.bit.BAUD = (VARIANT_MCK / ( 2 * 400000) - 1);
+    sercom3.enableWIRE();   //Re-enable I2C
```
Comment (fingolfin) [inline Adafruit_SSD1306.cpp]: I just checked, and `Wire.setClock` isn't a dummy function, at least in Arduino 1.6.7 with version 1.6.2 of the SAMD board kit.

File: Adafruit_SSD1306.cpp [hunk 2]
```diff
@@ -17,7 +17,7 @@ All text above, and the splash screen below must be included in any redistributi
 *********************************************************************/
 
 #include <avr/pgmspace.h>
-#ifndef __SAM3X8E__
+#if !defined(__SAM3X8E__) && !defined(__SAMD21G18A__)
```
Comment (fingolfin) [inline Adafruit_SSD1306.cpp]: PR #40 might be a better fix for this particular itch.

File: Adafruit_SSD1306.cpp [hunk 3]
```diff
@@ -506,7 +513,7 @@ void Adafruit_SSD1306::display(void) {
   else
   {
     // save I2C bitrate
-#ifndef __SAM3X8E__
+#if !defined(__SAM3X8E__) && !defined(__SAMD21G18A__)
     uint8_t twbrbackup = TWBR;
     TWBR = 12; // upgrade to 400KHz!
 #endif
```
Comment (fingolfin) [inline Adafruit_SSD1306.cpp]: The TWBR related changes in this PR should be not needed anymore in the current master branch.

File: Adafruit_SSD1306.cpp (hunk 1 above)
Comment (fingolfin) [inline Adafruit_SSD1306.cpp]: So I guess I want to ask: Is this part of the PR really (still?) necessary?

**Comment by fingolfin** [comment]
This PR does not apply anymore, and a lot of it should be obsolete by now, but it's hard to tell. Perhaps you can rebase it based on the current master branch, if there are any changes remaining for Zero compatibility?

See also my PR #44, which also attempts to fix usage of util/delay.h among other things

**Comment by fingolfin** [comment]
@jake-b Any chance you may want to comment on my comments? It would be good to know if this is still needed now that #44 has been merged, and if so, a rebased version of this would be helpful :)

**Comment by jake-b** [comment]
Hi Max.  Sorry for the delay.

I'm just going to withdraw this PR.  I no longer have the hardware setup to work on this, and it looks like the Arduino libraries have matured since then.  (setClock is definitely implemented now)

It was relevant when I originally sent the PR, but given in the intervening 6 months, it looks like further development has rendered it unnecessary.

**Comment by jake-b** [comment]
Withdrawn.

**Pull request closed without merging**